cluster-autoscaler/simulator/clustersnapshot/store/delta.go

```diff
@@ -241,6 +244,21 @@ func (data *internalDeltaSnapshotData) addPod(pod *apiv1.Pod, nodeName string) e
 	return nil
 }
 
+func (data *internalDeltaSnapshotData) addPodsForNode(pods []*apiv1.Pod, nodeName string) error {
+	ni, found := data.nodeInfoToModify(nodeName)
```

Comment:
This is brittle. `nodeInfoToModify` isn't safe to call concurrently: `modifiedNodeInfoMap` is going to be modified by multiple goroutines. This shouldn't happen when the entire snapshot is being initialized, but as the code evolves, it will be easy to miss this. Could you just pass `NodeInfo` directly here?

Reply:
That's right, changed